app/lib/synthetic_study_populator.rb

```diff
@@ -61,35 +61,25 @@ def self.add_files(study, study_config, synthetic_study_folder, user)
     file_infos = study_config['files']
     file_infos.each do |finfo|
       infile = File.open("#{synthetic_study_folder}/#{finfo['filename']}")
-      taxon_id = nil
-      if finfo['species_scientific_name'].present?
-        taxon = Taxon.find_by(scientific_name: finfo['species_scientific_name'])
-        if taxon.nil?
-          throw "You must populate the species #{finfo['species_scientific_name']} to ingest the file #{finfo['filename']}. Stopping populate"
-        end
-        taxon_id = taxon.id
-      end
 
       study_file_params = {
         file_type: finfo['type'],
         name: finfo['name'] ? finfo['name'] : finfo['filename'],
         upload: infile,
         use_metadata_convention: finfo['use_metadata_convention'] ? true : false,
         status: 'uploading',
-        study: study,
-        taxon_id: taxon_id
+        study: study
       }
 
+      study_file_params = study_file_params.merge(process_genomic_file_params(finfo))
```

Comment:
Small nit - you can call `study_file_params.merge!(process_genomic_file_params(finfo))` instead.

Reply:
oooh, nice, I hadn't used that Ruby method before